google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -162,16 +161,13 @@ protected void hideWarning() {
       public void run() {
         cloudSdkDirectoryField.getTextField().setForeground(JBColor.black);
         warningIcon.setVisible(false);
-        warningIcon.setVisible(false);
         warningMessage.setVisible(false);
-        cloudSdkPanel.updateUI();
       }
     });
   }
 
   private void invokePanelValidationUpdate(Runnable runnable) {
-    ApplicationManager.getApplication().invokeLater(runnable,
-        ModalityState.stateForComponent(cloudSdkPanel));
+    ApplicationManager.getApplication().invokeLater(runnable, ModalityState.current());
```

Comment:
Just to make sure I understand:

You said that "It doesn't prevent from saving the dialog". So before your change, there was no warning and the user could save the dialog (even if it was in an invalid state). Now, the user sees the warning but can save the dialog anyway? I think the behavior we want is to see the warning and no be able to save from those dialogs that require a valid SDK, but allow saving from those that don't (I'm not sure if this is what you meant that is happening).